packages/react/src/SubdomainNavBar/SubdomainNavBar.tsx

```diff
@@ -81,11 +82,21 @@ function Root({
   const [menuHidden, setMenuHidden] = useState(true)
   const [searchVisible, setSearchVisible] = useState(false)
   const {isSmall, isMedium} = useWindowSize()
+  const [startOfContentButtonFocused, setStartOfContentButtonFocused] = useState(false)
+  const [mainTag, setMainTag] = useState<HTMLElement | undefined>(undefined)
+  const startOfContentID = useId('start-of-content')
 
   const handleMobileMenuClick = () => setMenuHidden(!menuHidden)
   const handleSearchVisibility = () => setSearchVisible(!searchVisible)
   const focusTrapRef = useRef<HTMLDivElement | null>(null)
 
+  useEffect(() => {
+    const main = document.getElementsByTagName('main')[0] as HTMLElement | undefined
+    if (!main) return
+    if (!main.id && startOfContentID) main.id = startOfContentID
+    setMainTag(main)
+  }, [setMainTag, startOfContentID])
```

Comment:
@rezrah this seems... somehow wrong, but unsure how else I might satisfy the a11y team's recommendation. Do you have any thoughts?

Reply:
I wouldn't recommend storing the element in state, if that's what you're asking?That's really heavy and `refs` are better for this IMO.

This is just pseudocode so you'll need to try it, but can you do something like:

```jsx
const mainElRef = useRef<HTMLElement | null>(null);

useEffect(() => {
    const main = document.getElementsByTagName('main')[0] as HTMLElement | undefined
    
  if(el and otherStuff) {
    mainElRef.current = el;
  }

}, []);
```

Reply:
@rezrah my man, I think this is GTG ☺️ , except these [two testing stories](https://github.com/primer/brand/blob/ddd5b28e4aee6835c25b4b2681768bda29987214/packages/react/src/SubdomainNavBar/SubdomainNavBar.stories.tsx#L643-L690) related to this update are giving AXE errors in CI! Would you know why it is failing? My AXE tooling is showing 0 issues even on strict. And running `test:axe` doesn't work on Codespaces, so I'll have to try this on my local.